system/jlib/jflz.cpp

```diff
@@ -723,11 +732,28 @@ class jlib_decl CFastLZCompressor : public CInterface, public ICompressor
 
     size32_t write(const void *buf,size32_t len)
     {
+        // fill internal buffer with smaller chunks for better fit/less gap fill
+        size32_t lenb = wrmax;
+        if (len > lenb)
+        {
+            byte *b = (byte *)buf;
+            size32_t lentot = 0;
+            while (lentot < len)
+            {
+                if ((lentot + lenb) > len)
+                    lenb = len - lentot;
+                size32_t amnt = write((const void *)(b+lentot), lenb);
```

Comment:
to my mind, this isn't a natural use of recursion.
It will only ever recurse once, and fall out of into the block outside of this block as (len > lenb) will not be true.

It looks like this could be simplified as one while loop without a call, something like:
```code
while (len)
{
         if (len < lenb)
              lenb = len;
         if (lenb+inlen>inmax)
         {
             if (trailing)
                 return ... <amount written so far> (b-buf?);
             flushcommitted();
             if (lenb+inlen>inmax)
                  lenb = inmax-inlen;
         }
         memcpy(inbuf+inlen,b,len);
         b += lenb;
         inlen += lenb;
         len -= lenb;
}
```